### DISCUSSION TITLE
Crop not respecting gravity?

### DISCUSSION THREAD

**Discussion by Gatada** [ImageMagick/ImageMagick#4557] [Help] [answered]
I'm trying to improve the dithering of some images that will be shown on an eInk 4bit display.

For this I am trying the following:

`convert gray.jpg -resize 2560x1440^ -gravity center -crop 2560x1440! -dither FloydSteinberg -colors 16 -grayscale Lightness -define dither:diffusion-amount=20% grayOut.png`

I'm not sure why, but I have the following issues:
1. The cropping does not retain the center of the image; I thought ```-gravity center``` would shave off the excess symmetrically?
2. I get two "tiles" as the cropped part of the image is retained in a new file. I do not want this, and thought using crop "a single image is generated"?
3. Using ```-define dither:diffusion-amount=0%``` or ```-define dither:diffusion-amount=100%``` results in optically identical results, what am I missing?
4. Can I somehow avoid this warning? ```convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `grayOut2.png' @ warning/png.c/MagickPNGWarningHandler/1748.```

Thanks!

**Comment by fmw42** [comment, marked as answer]
1) Cropping will take the center 2560x1440. Those sizes are relative to the resized image, not the input image.

2) Your get two tiles because you have not specified a +X+Y for your crop. Try -crop 2560x1440+0+0. I do not think you want to use ! here.

3) I do not know why they produce the same results

4) ImageMagick will produce this warning, when you save a png from a grayscale image. Try adding -quiet after convert and before reading the input to suppress warnings.

**Reply by Gatada** [reply]
Thanks, all issues solved! I do find it curious though, that without an offset, the crop operation doesn't respect gravity? I even tried ```diffusion-amount=30000%``` with no discernible effect.

**Comment by fmw42** [comment]
Without an offset, -crop tiles the image into as many tiles as it can find of that size. See https://legacy.imagemagick.org/Usage/crop/#crop_tile and https://legacy.imagemagick.org/Usage/crop/#crop_tile_centered

**Comment by snibgo** [comment]
Q3: You have have "-define dither:diffusion-amount=20%" _after_ you have done the dithering in 2-colors", so that is too late. If it has an effect, it must come _before_ the dithering.

What version of IM, on what platform? Perhaps "-define dither:diffusion-amount=20%" wasn't implemented in your version.

If you use IM V7, I suggest "magick" instead of "convert".

Cropping does respect gravity. Insert "+repage" after the crop, eg "-gravity SouthEast -crop 200x100+0+0 +repage".

**Reply by fmw42** [reply]
When saving to PNG, you would need +repage after the crop. But if saving to JPG, that would not be needed as PNG supports saving the virtual canvas information and JPG does not.
